lib/api/utilities/resource_link_parser.rb

```diff
@@ -30,20 +30,60 @@
 module API
   module Utilities
     module ResourceLinkParser
-      def self.parse(resource_link)
-        ::API::Root.routes.each do |route|
-          route_options = route.instance_variable_get(:@options)
-          match = route_options[:compiled].match(resource_link)
-
-          if match
-            return {
-              ns: /\/(?<ns>\w+)\//.match(route_options[:namespace])[:ns],
-              id: match[:id]
-            }
+      # N.B. valid characters for URL path segments as of
+      # http://tools.ietf.org/html/rfc3986#section-3.3
+      SEGMENT_CHARACTER = '(\w|[-~!$&\'\(\)*+\.,:;=@]|%[0-9A-Fa-f]{2})'
+      RESOURCE_REGEX = "/api/v(?<version>\\d)/(?<namespace>\\w+)/(?<id>#{SEGMENT_CHARACTER}*)\\z"
+
+      class << self
+        def parse(resource_link)
+          match = resource_matcher.match(resource_link)
+
+          return nil unless match
+
+          return {
+            version: match[:version],
+            namespace: match[:namespace],
+            id: match[:id]
+          }
+        end
+
+        def parse_id(resource_link,
```

Comment:
Assignment Branch Condition size for parse_id is too high. [16.4/15]<br>Perceived complexity for parse_id is too high. [9/7]